src/java.base/share/classes/sun/security/ssl/ClientHello.java

```diff
@@ -538,14 +535,6 @@ private ClientHelloKickstartProducer() {
                 if (!session.getProtocolVersion().useTLS13PlusSpec()) {
                     sessionId = session.getSessionId();
                 }
-                if (!maxProtocolVersion.equals(sessionVersion)) {
-                    maxProtocolVersion = sessionVersion;
-
-                    // Update protocol version number in underlying socket and
-                    // handshake output stream, so that the output records
-                    // (at the record layer) have the correct version
-                    chc.setVersion(sessionVersion);
```

Comment:
The removing of the call to "setVersion()" has an impact, I think.  I think the declaration of this method could be removed in HandshakeContext class, and set the HandshakeContext.conContext.protocolVersion to HandshakeContext.maximumActiveProtocol in the  HandshakeContext.initialize() method.

Reply:
I've removed the (now) redundant setVersion(..) method.

With respect to initialize() method we already set  HandshakeContext.conContext.protocolVersion  to HandshakeContext.maximumActiveProtocol in the case of a new session. Is that what you were aiming at ?
https://github.com/openjdk/jdk/blob/1ea437a4b87381b558cf8157ac52fc03e37ac507/src/java.base/share/classes/sun/security/ssl/HandshakeContext.java#L261

Reply:
Yes.  Thank you!